tracex_project/extraction/logic/modules/module_time_extractor.py

```diff
@@ -102,10 +102,24 @@ def fix_end_dates(row):
 
         df["time:timestamp"] = pd.to_datetime(
             df["time:timestamp"], format="%Y%m%dT%H%M", errors="coerce"
-        ).ffill()
+        )
         df["time:end_timestamp"] = pd.to_datetime(
             df["time:end_timestamp"], format="%Y%m%dT%H%M", errors="coerce"
-        ).ffill()
+        )
+
+        if df["time:timestamp"].isna().all():
+            df["time:timestamp"] = df["time:timestamp"].fillna(
+                pd.Timestamp("2020-01-01 00:00")
+            )
+
+        if df["time:end_timestamp"].isna().all():
+            df["time:end_timestamp"] = df["time:end_timestamp"].fillna(
+                pd.Timestamp("2020-01-01 00:00")
+            )
+
+        df["time:timestamp"] = df["time:timestamp"].bfill().ffill()
+        df["time:end_timestamp"] = df["time:end_timestamp"].bfill().ffill()
```

Comment:
Lets discuss the order of filling up the missing values. My gut says timestamps are more likely in the beginning of a patient journey than at the end. Also phrases like "after that" are more likely than "before that". So I suggest using `ffill()` and then `bfill()`. Opinions?

Reply:
Thats true, it was on my mind too, but it slipped. Thanks!